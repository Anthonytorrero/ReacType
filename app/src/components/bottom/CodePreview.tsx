import React, { useContext } from 'react';
import { stateContext } from '../../context/context';
import AceEditor from 'react-ace';
import { makeStyles } from '@material-ui/core/styles';
import 'ace-builds/src-noconflict/mode-javascript';
import 'ace-builds/src-noconflict/theme-monokai';
import Button from '@material-ui/core/Button';
import { Component } from '../../interfaces/Interfaces';

const CodePreview = () => {
  const [state, dispatch] = useContext(stateContext);
  const currentComponent = state.components.find(
    (elem: Component) => elem.id === state.canvasFocus.componentId
  );

  return (
    <div
      style={{
        height: '90%',
        paddingLeft: '0px',
        paddingTop: '1rem',
        paddingBottom: '1rem',
        overflow: 'auto',
        maxWidth: '100%',
        display: 'flex',
        justifyContent: 'center'
      }}
    >
      
      <AceEditor
        mode="javascript"
        theme="monokai"
        width="100%"
        height="100%"
        style={{
          border: '2px solid #33eb91',
          borderRadius: '8px'
        }}
        // onChange={code =>
        //   this.props.updateCode({
        //     componentId: this.props.focusComponent.id,
        //     code
        //   })
        // }
        value={currentComponent.code}
        name="Code_div"
        // readOnly={this.props.codeReadOnly}
        readOnly={true}
        editorProps={{ $blockScrolling: true }}
        fontSize={16}
        tabSize={2}
      />
    </div>
  );
};

const useStyles = makeStyles(theme => ({}));

export default CodePreview;
