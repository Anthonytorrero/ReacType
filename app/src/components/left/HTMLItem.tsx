import React from 'react';
import { useDrag } from 'react-dnd';
import { ItemTypes } from '../../constants/ItemTypes';
import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles({
  HTMLPanelItem: {
    color: 'white',
    // this is experimental for version: BLADERUNNER THEME
    backgroundColor: 'transparent',
    // minWidth: '340px',
    minHeight: '60px',
    marginBottom: '10px',
    marginRight: '5px',
    marginLeft: '5px',
    border: '2px dotted rgba(255,255,255, 0.45)',
    borderRadius: '8px',
    cursor: 'grab',
    '& > h3': {
      display: 'inline-block',
      paddingTop: '18px'
    }
  }
});

const HTMLItem: React.FC<{
  name: string;
  id: number;
  Icon: any;
}> = ({ name, id, Icon }) => {
  const classes = useStyles();

  const [{ isDragging }, drag] = useDrag({
    item: {
      type: ItemTypes.INSTANCE,
      newInstance: true,
      instanceType: 'HTML Element',
      name,
      instanceTypeId: id
    },
    collect: (monitor: any) => ({
      isDragging: !!monitor.isDragging()
    })
  });

  return (
    <Grid item xs={5} key={`html-${name}`}>
      <div ref={drag} className={classes.HTMLPanelItem}>
        <h3>{name}</h3>
        <span
          style={{
            verticalAlign: 'middle',
            display: 'inline-block',
            marginLeft: '5px'
          }}
        >
          {Icon && <Icon />}
        </span>
      </div>
    </Grid>
  );
}

export default HTMLItem;
