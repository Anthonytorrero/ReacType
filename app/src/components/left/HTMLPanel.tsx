import React from 'react';
import Grid from '@material-ui/core/Grid';
import { useDrag } from 'react-dnd';
import { ItemTypes } from '../../constants/ItemTypes';
import HTMLTypes from '../../context/HTMLTypes';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
  HTMLPanelItem: {
    color: 'white',
    // this is experimental for version: BLADERUNNER THEME
    backgroundColor: 'transparent',
    // minWidth: '340px',
    minHeight: '60px',
    marginBottom: '10px',
    marginRight: '5px',
    marginLeft: '5px',
    border: '2px dotted rgba(255,255,255, 0.45)',
    borderRadius: '8px',
    cursor: 'grab',
    '& > h3': {
      display: 'inline-block',
      paddingTop: '18px'
    }
  }
});

const HTMLPanel = (): JSX.Element => {
  const classes = useStyles();
  const options = HTMLTypes.map(option => {
    const [{ isDragging }, drag] = useDrag({
      item: {
        type: ItemTypes.INSTANCE,
        newInstance: true,
        instanceType: 'HTML Element',
        name: option.name,
        instanceTypeId: option.id
      },
      collect: (monitor: any) => ({
        isDragging: !!monitor.isDragging()
      })
    });
    return (
      <Grid item xs={5} key={`html-${option.name}`}>
        <div ref={drag} className={classes.HTMLPanelItem}>
          <h3>{option.name}</h3>
          <span
            style={{
              verticalAlign: 'middle',
              display: 'inline-block',
              marginLeft: '5px'
            }}
          >
            {<option.icon />}
          </span>
        </div>
      </Grid>
    );
  });

  return (
    <div>
      <h4> HTML Elements</h4>
      <Grid
        container
        spacing={1}
        direction="row"
        justify="center"
        alignItems="center"
      >
        {options}
      </Grid>
    </div>
  );
};

export default HTMLPanel;
