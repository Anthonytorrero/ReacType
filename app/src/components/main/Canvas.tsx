import React, { useState, useContext } from 'react';
import { useDrop, DropTargetMonitor } from 'react-dnd';
import _ from 'lodash';
import { ItemTypes } from '../../constants/ItemTypes';
import StateContext from '../../context/context';
import { Component, DragItem } from '../../interfaces/Interfaces';
import { combineStyles } from '../../helperFunctions/combineStyles';
import renderChildren from '../../helperFunctions/renderChildren';

// const snapStateArr = [];
function Canvas() {
  const [state, dispatch] = useContext(StateContext);
  console.log('state in Canvas', state);
  // const [ prevState, setPrevState ] = useState<Array<object>>([]); // NOT USING
  // find the current component to render on the canvas
  const currentComponent: Component = state.components.find(
    (elem: Component) => elem.id === state.canvasFocus.componentId
    );

  // console.log('currentComponent', currentComponent)

  // changes focus of the canvas to a new component / child
  const changeFocus = (componentId: number, childId: number | null) => {
    dispatch({ type: 'CHANGE FOCUS', payload: { componentId, childId } });
    
  };
  // onClickHandler is responsible for changing the focused component and child component
  function onClickHandler(event) {
    event.stopPropagation();
    // note: a null value for the child id means that we are focusing on the top-level component rather than any child
    changeFocus(state.canvasFocus.componentId, null);
  }

  // stores a snapshot of state into the past array for UNDO
  const snapShotFunc = () => {
    // make a deep clone of state
      const deepCopiedState = JSON.parse(JSON.stringify(state));
      state.past.push(deepCopiedState.components[0].children);
      // console.log('state in snapShotFunc', state.past)
    }

  
  // This hook will allow the user to drag items from the left panel on to the canvas
  const [{ isOver }, drop] = useDrop({
    accept: ItemTypes.INSTANCE,
    drop: (item: DragItem, monitor: DropTargetMonitor) => {
      const didDrop = monitor.didDrop();
      // returns false for direct drop target
      snapShotFunc();
      if (didDrop) {
        return;
      }
      // console.log('item', item)
      // if item dropped is going to be a new instance (i.e. it came from the left panel), then create a new child component
      if (item.newInstance) {
        dispatch({
          type: 'ADD CHILD',
          payload: {
            type: item.instanceType,
            typeId: item.instanceTypeId,
            childId: null
          }
        });
      }
      // if item is not a new instance, change position of element dragged inside div so that the div is the new parent
      else {
        dispatch({
          type: 'CHANGE POSITION',
          payload: {
            // name: item.name,
            currentChildId: item.childId,
            newParentChildId: null
          }
        });
      }
    },
    collect: monitor => ({
      isOver: !!monitor.isOver()
    })
  });

  // Styling for Canvas
  const defaultCanvasStyle = {
    width: '100%',
    minHeight: '100%',
    backgroundColor: isOver ? 'lightyellow' : '#F5F5F5',
    backgroundImage: "url('https://www.transparenttextures.com/patterns/diagonal-noise.png')",
    border: '1px solid #FBFBF2',
    borderStyle: isOver ? 'dotted' : 'solid'
  };
  // Combine the default styles of the canvas with the custom styles set by the user for that component
  // The render children function renders all direct children of a given component
  // Direct children are draggable/clickable
  const canvasStyle = combineStyles(defaultCanvasStyle, currentComponent.style);
  return (
    <div ref={drop} style={canvasStyle} onClick={onClickHandler}>
      {/* currentComponent is the selected component on Left Panel (eg: App or Index with green dot to the left)  */}
      {renderChildren(currentComponent.children)}
    </div>
  );
}
// export { snapStateArr };
export default Canvas;

