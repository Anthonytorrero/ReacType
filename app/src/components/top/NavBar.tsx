import React, { useState, useContext } from 'react';
import {
  withStyles,
  createStyles,
  makeStyles,
  Theme
} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import { styleContext } from '../../containers/AppContainer';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import LoginButton from '../right/LoginButton'
import createModal from '../right/createModal';
import StateContext from '../../context/context';

// NavBar text and button styling
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      width: '100%'
    },
    menuButton: {
      marginRight: theme.spacing(2),
      color: 'white'
    },
    title: {
      flexGrow: 1,
      color: 'white'
    }
  })
);

// Drop down menu button for export
const StyledMenu = withStyles({
  paper: {
    border: '1px solid #d3d4d5'
  }
})(props => (
  <Menu
    elevation={0}
    getContentAnchorEl={null}
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center'
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'center'
    }}
    {...props}
  />
));

const StyledMenuItem = withStyles(theme => ({
  root: {
    '&:focus': {
      backgroundColor: theme.palette.primary.main,
      '& .MuiListItemIcon-root, & .MuiListItemText-primary': {
        color: theme.palette.common.white
      }
    }
  }
}))(MenuItem);

export default function NavBar(props) {
  const classes = useStyles();

  const { style, setStyle } = useContext(styleContext);

  // State for export menu button
  const [anchorEl, setAnchorEl] = React.useState(null);
  // State for clear canvas button
  const [modal, setModal] = useState(null);
  const [state, dispatch] = useContext(StateContext);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  // ---Clear canvas functionality---
  // Closes out the open modal
  const closeModal = () => setModal('');

  // Creates modal that asks if user wants to clear workspace
  // If user clears their workspace, then their components are removed from state and the modal is closed
  const clearWorkspace = () => {
    // Reset state for project to initial state
    const resetState = () => {
      dispatch({ type: 'RESET STATE', payload: {} });
    };

    // Set modal options
    const children = (
      <List className="export-preference">
        <ListItem
          key={'clear'}
          button
          onClick={resetState}
          style={{
            border: '1px solid #3f51b5',
            marginBottom: '2%',
            marginTop: '5%'
          }}
        >
          <ListItemText
            primary={'Yes, delete all project data'}
            style={{ textAlign: 'center' }}
            onClick={closeModal}
          />
        </ListItem>
      </List>
    );

    // Create modal
    setModal(
      createModal({
        closeModal,
        children,
        message: 'Are you sure want to delete all data?',
        primBtnLabel: null,
        primBtnAction: null,
        secBtnAction: null,
        secBtnLabel: null,
        open: true
      })
    );
  };

  // added className="navbarButton" to test for buttons in enzyme.test.tsx
  return (
    <div className={classes.root} style={style}>
      <AppBar position="static">
        <Toolbar>
          <IconButton
            edge="start"
            className={classes.menuButton}
            color="inherit"
            aria-label="menu"
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" className={classes.title}>
            ReactType
          </Typography>
          <Button 
            variant="contained"
            color="primary"
            style={{minWidth: '137.69px'}}
            onClick={clearWorkspace} 
            className="navbarButton" 
            id="navbarButton"
          >
            Clear Canvas
          </Button>
          <Button
            // aria-controls="customized-menu"
            // aria-haspopup="true"
            variant="contained"
            color="primary"
            onClick={handleClick}
            className="navbarButton"
            id="navbarButton"
          >
            Export
          </Button>

          <StyledMenu
            id="customized-menu"
            anchorEl={anchorEl}
            keepMounted
            open={Boolean(anchorEl)}
            onClose={handleClose}
          >
            <StyledMenuItem>
              <ListItemText primary="Classic React" />
            </StyledMenuItem>

            <StyledMenuItem>
              <ListItemText primary="Next.js" />
            </StyledMenuItem>

            <StyledMenuItem>
              <ListItemText primary="Gatsby" />
            </StyledMenuItem>
          </StyledMenu>
          <Button
            className="navbarButton"
            id="navbarButton"
            color="primary"
            variant="contained"
            style={{minWidth: '113.97px'}}
            onClick={() => {
              !style.backgroundColor
                ? setStyle({ backgroundColor: '#21262D' }) //dark mode color
                : setStyle({});
              props.isThemeLight ? props.setTheme(false) : props.setTheme(true);
            }}
          >
            Dark Mode
          </Button>
          {/* <Button color="inherit" className="navbarButton">Login</Button> */}
          <LoginButton />
        </Toolbar>
      </AppBar>
      {modal}
    </div>
  );
}
