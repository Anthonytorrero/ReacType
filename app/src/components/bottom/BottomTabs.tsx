import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
// import Tree from 'react-d3-tree';
import CodePreview from './CodePreview';
import Box from '@material-ui/core/Box';
import { emitKeypressEvents } from 'readline';

const BottomTabs = () => {
  // state that controls which tab the user is on
  const [tab, setTab] = useState(0);
  const classes = useStyles();
  treeWrapper: HTMLDivElement;

  // const el = document.querySelector('resizer');
  // console.log(el);
  // el.addEventListener('mousedown', mousedown);

  // function mousedown(e) {
  //   window.addEventListener('mousemove', mousemove);
  //   window.addEventListener('mouseup', mouseup);

  //   let prevX = e.clientX;
  //   let prevY = e.clientY;

  //   function mousemove(e) {
  //     let newX = prevX - e.clientX;
  //     let newY = prevY - e.clientY;

  //     const rect = el.getBoundingClientRect();

  //     el.style.left = rect.left - newX + 'px';
  //     el.style.top = rect.top - newY + 'px';

  //     prevX = e.clientX;
  //     prevY = e.clientY;
  //   }
  //   function mouseup() {
  //     window.removeEventListener('mousemove', mousemove);
  //     window.removeEventListener('mouseup', mouseup);
  //   }
  // }

  // method changes the
  const handleChange = (event: React.ChangeEvent, value: number) => {
    setTab(value);
  };

  return (
    <div className={classes.root}>
      {/* <div id="resizeMe" className="resizable"> */}
      <Box display="flex" justifyContent="space-between">
        <Tabs
          value={tab}
          onChange={handleChange}
          classes={{
            root: classes.tabsRoot,
            indicator: classes.tabsIndicator
          }}
        >
          <Tab
            disableRipple
            classes={{ root: classes.tabRoot, selected: classes.tabSelected }}
            label="Code Preview"
          />
          <Tab
            classes={{
              root: classes.tabsRoot,
              selected: classes.tabSelected
            }}
            label="Tree Diagram"
          />
        </Tabs>
      </Box>
      {tab === 0 && <CodePreview />}
      {/* </div> */}
    </div>
  );
};

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    backgroundColor: '#333333',
    height: '100%',
    color: '#fff',
    boxShadow: '0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)'
  },
  bottomHeader: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    Width: '200px'
  },
  tabsRoot: {
    borderBottom: '0.5px solid #424242'
  },
  tabsIndicator: {
    backgroundColor: '#1de9b6'
  },
  tabRoot: {
    textTransform: 'initial',
    minWidth: 40,
    fontWeight: theme.typography.fontWeightRegular,
    marginRight: theme.spacing(4), // JZ: updated syntax as per deprecation warning

    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      '"Helvetica Neue"',
      'Arial',
      'sans-serif',
      '"Apple Color Emoji"',
      '"Segoe UI Emoji"',
      '"Segoe UI Symbol"'
    ].join(','),
    '&:hover': {
      color: '#1de9b6',
      opacity: 1
    },
    '&$tabSelected': {
      color: '#33eb91',
      fontWeight: theme.typography.fontWeightMedium
    },
    '&:focus': {
      color: '#4aedc4'
    }
  },
  tabSelected: {},
  typography: {
    padding: theme.spacing(3) // JZ: updated syntax as per deprecation warning
  },
  padding: {
    padding: `0 ${theme.spacing(2)}px` // JZ: updated syntax as per deprecation warning
  },
  switch: {
    marginRight: '10px',
    marginTop: '2px'
  }
}));

export default BottomTabs;
