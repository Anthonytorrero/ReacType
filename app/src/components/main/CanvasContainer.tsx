import React from 'react';
import Canvas from './Canvas';

// The CanvasContainer sets the boundaries on the width/height of the canvas
function CanvasContainer() {
  const canvasContainerStyle = {
    height: '950px',
    width: '100%',
    backgroundColor: 'lightgrey',
    border: '2px Solid grey',
    overflowY: 'auto'
  };

  return (
    <div style={canvasContainerStyle}>
      <Canvas />
    </div>
  );
}

export default CanvasContainer;
