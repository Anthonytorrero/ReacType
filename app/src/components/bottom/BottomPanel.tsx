import React, { useContext } from 'react';
import { stateContext } from '../../context/context';
import BottomTabs from './BottomTabs';

// const IPC = require('electron').ipcRenderer;

const BottomPanel = () => {
  return (
    <div className="bottom-panel" style={{ width: '100%', height: '100%' }}>
      <BottomTabs />
    </div>
  );
};

export default BottomPanel;