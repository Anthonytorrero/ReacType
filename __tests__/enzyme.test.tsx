import { shallow, mount } from 'enzyme';
import React, { useState, useContext } from 'react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import StateContext from '../app/src/context/context';
import initialState from '../app/src/context/initialState';
import MainContainer from '../app/src/containers/MainContainer';
import BottomPanel from '../app/src/components/bottom/BottomPanel';
import BottomTabs from '../app/src/components/bottom/BottomTabs';
import CodePreview from '../app/src/components/bottom/CodePreview';
import CanvasContainer from '../app/src/components/main/CanvasContainer';
import Canvas from '../app/src/components/main/Canvas';
import HTMLPanel from '../app/src/components/left/HTMLPanel';
import HTMLItem from '../app/src/components/left/HTMLItem';
import LeftContainer from '../app/src/containers/LeftContainer';
import AppContainer from '../app/src/containers/AppContainer';
import NavBar from '../app/src/components/top/NavBar';

// npm test -- -u

describe('Test the CanvasContainer component', () => {
  const target = shallow(<CanvasContainer />);
  it('Matches snapshot', () => {
    expect(target).toMatchSnapshot();
  });
  it('Contains Canvas component', () => {
    expect(target.contains(<Canvas />)).toBe(true);
  });
});

describe('Test the MainContainer component', () => {
  const target = shallow(<MainContainer />);
  it('Contains CanvasContainer component', () => {
    expect(target.contains(<CanvasContainer />)).toBe(true);
  });
  it('Contains BottomPanel component', () => {
    expect(target.contains(<BottomPanel />)).toBe(true);
  });
});

describe('Test the BottomTabs component', () => {
  const target = shallow(<BottomTabs />);
  it('Matches snapshot', () => {
    expect(target).toMatchSnapshot();
  });
});

describe('Test All 10 default HTML elements have rendered', () => {
  const target = shallow(
    <DndProvider backend={HTML5Backend}>
      <StateContext.Provider value={initialState}>
        <HTMLPanel />
      </StateContext.Provider>
    </DndProvider>
  );

  it('Matches snapshot', () => {
    expect(target).toMatchSnapshot();
  });
});

// testing for AppContainer
describe('Test AppContainer container', () => {
  const target = shallow(<AppContainer />);

  const props = {
    setTheme: jest.fn(),
    isThemeLight: jest.fn()
  };

  // testing for a NavBar
  it('Should render NavBar', () => {
    expect(
      target.find(
        <NavBar setTheme={props.setTheme} isThemeLight={props.isThemeLight} />
      )
    ).toBeDefined();
  });
});
// testing for NavBar component
describe('Test NavBar component', () => {
  const props = {
    setTheme: jest.fn(),
    isThemeLight: jest.fn()
  };
  const target = shallow(
    <NavBar setTheme={props.setTheme} isThemeLight={props.isThemeLight} />
  );
  console.log('array', target.find('.navbarButton'));
  // testing for 4 generic buttons in NavBar
  it('Should render 4 buttons: "Clear Canvas", "Export", "Dark Mode", "Login"', () => {
    expect(target.find('.navbarButton')).toHaveLength(4);
    expect(
      target
        .find('.navbarButton')
        .at(0)
        .text()
    ).toEqual('Clear Canvas');
     expect(
       target
         .find('.navbarButton')
         .at(1)
         .text()
     ).toEqual('Export');
      expect(
      target
        .find('.navbarButton')
        .at(2)
        .text()
    ).toEqual('Dark Mode');
     expect(
       target
         .find('.navbarButton')
         .at(3)
         .text()
     ).toEqual('Login');
  });
});
