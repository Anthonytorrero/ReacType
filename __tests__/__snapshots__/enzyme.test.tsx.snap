// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the BottomTabs component Matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
  style={null}
>
  <Styled(MuiBox)
    display="flex"
    justifyContent="space-between"
  >
    <WithStyles(ForwardRef(Tabs))
      classes={
        Object {
          "indicator": "makeStyles-tabsIndicator-4",
          "root": "makeStyles-tabsRoot-3",
        }
      }
      onChange={[Function]}
      value={0}
    >
      <WithStyles(ForwardRef(Tab))
        classes={
          Object {
            "root": "makeStyles-tabRoot-5",
            "selected": "makeStyles-tabSelected-6",
          }
        }
        disableRipple={true}
        label="Code Preview"
      />
      <WithStyles(ForwardRef(Tab))
        classes={
          Object {
            "root": "makeStyles-tabRoot-5",
            "selected": "makeStyles-tabSelected-6",
          }
        }
        disableRipple={true}
        label="Component Tree"
      />
    </WithStyles(ForwardRef(Tabs))>
    <WithStyles(ForwardRef(FormControl))>
      <div
        className="flex-container"
      >
        <div
          className="flex1"
        >
          Change Theme:
        </div>
        <WithStyles(ForwardRef(NativeSelect))
          className="flex2"
          onChange={[Function]}
          style={
            Object {
              "color": "white",
            }
          }
          value="monokai"
        >
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="monokai"
          >
            Monokai
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="github"
          >
            Github
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="solarized_dark"
          >
            Solarized Dark
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="terminal"
          >
            Terminal
          </option>
          <option
            style={
              Object {
                "backgroundColor": "#252526",
              }
            }
            value="solarized_light"
          >
            Solarized Light
          </option>
        </WithStyles(ForwardRef(NativeSelect))>
      </div>
    </WithStyles(ForwardRef(FormControl))>
  </Styled(MuiBox)>
  <CodePreview
    setTheme={[Function]}
    theme="monokai"
  />
</div>
`;

exports[`Test the CanvasContainer component Matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "lightgrey",
      "border": "2px Solid grey",
      "width": "100%",
    }
  }
>
  <Canvas />
</div>
`;
